kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiStatement.java

```diff
@@ -743,4 +763,57 @@ public String getQueryId() throws SQLException {
   public void setInPlaceUpdateStream(InPlaceUpdateStream stream) {
     this.inPlaceUpdateStream = stream;
   }
+
+  private TGetResultSetMetadataResp getResultSetMetadata() throws SQLException {
+    try {
+      TGetResultSetMetadataReq metadataReq = new TGetResultSetMetadataReq(stmtHandle);
+      // TODO need session handle
+      TGetResultSetMetadataResp metadataResp;
+      metadataResp = client.GetResultSetMetadata(metadataReq);
+      Utils.verifySuccess(metadataResp.getStatus());
+      return metadataResp;
+    } catch (SQLException eS) {
+      throw eS; // rethrow the SQLException as is
+    } catch (Exception ex) {
+      throw new KyuubiSQLException("Could not create ResultSet: " + ex.getMessage(), ex);
+    }
+  }
+
+  private void parseMetadata(
+      TGetResultSetMetadataResp metadataResp,
+      List<String> columnNames,
+      List<TTypeId> columnTypes,
+      List<JdbcColumnAttributes> columnAttributes)
+      throws KyuubiSQLException {
+    TTableSchema schema = metadataResp.getSchema();
+    if (schema == null || !schema.isSetColumns()) {
+      throw new KyuubiSQLException("the result set schema is null");
+    }
+
+    // parse kyuubi hint
+    List<String> infoMessages = metadataResp.getStatus().getInfoMessages();
+    if (infoMessages != null) {
+      metadataResp
+          .getStatus()
+          .getInfoMessages()
+          .forEach(
+              line -> {
+                String[] keyValue = line.split("=");
+                String key = keyValue[0];
+                String value = keyValue[1];
```

Comment:
will cause NPE if server return some message does not contains '='

Reply:
Log a warning message if the input does not contain "="